src/main/java/fi/metatavu/dcfb/server/search/searchers/ItemSearcher.java

```diff
@@ -0,0 +1,100 @@
+package fi.metatavu.dcfb.server.search.searchers;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.index.query.QueryBuilders.queryStringQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchAllQuery;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.search.sort.SortBuilder;
+import org.elasticsearch.search.sort.SortBuilders;
+import org.elasticsearch.search.sort.SortOrder;
+
+import fi.metatavu.dcfb.server.rest.model.ItemListSort;
+import fi.metatavu.dcfb.server.search.index.IndexableItem;
+
+/**
+ * Searcher for items
+ */
+@ApplicationScoped
+public class ItemSearcher extends AbstractSearcher {
+
+  /**
+   * Searches item and returns result as UUIDs
+   * 
+   * @param categoryIds category ids that must exist on the result. Omitted if null
+   * @param search free text search that must match the result. Omitted if null
+   * @param firstResult first result. Defaults to 0
+   * @param maxResults max results. Defaults to 20
+   * @return search result 
+   */
+  public SearchResult<UUID> searchItems(List<UUID> categoryIds, String search, Long firstResult, Long maxResults, List<ItemListSort> sorts) {
+    boolean matchAll = categoryIds == null && search == null;
+    if (matchAll) {
+      return executeSearch(matchAllQuery(), createSorts(sorts), firstResult, maxResults);
+    } else {    
+      BoolQueryBuilder query = boolQuery();
+      
+      if (categoryIds != null) {
+        BoolQueryBuilder matchOrQuery = boolQuery();
+
+        categoryIds.forEach(categoryId -> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove useless curly braces around statement ([squid:S1602](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS1602))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.dcfb-api%3Adcfb-api&pullRequest=3&issues=AWQWmPQAbXXTsx7O6YI7&open=AWQWmPQAbXXTsx7O6YI7)

[ISSUE_KEY]: AWQWmPQAbXXTsx7O6YI7